init_process.go

```diff
@@ -0,0 +1,406 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+// static file path
+const ConfigFileFolderPath = "init_process_fluent-bit_s3_config_files"
+const MainConfigFilePath = "init_process_fluent-bit.conf"
+const OriginalMainConfigFilePath = "/fluent-bit/etc/fluent-bit.conf"
+const InvokerFilePath = "invoker.sh"
+
+// default Fluent Bit command
+var FluentBitCommand = "exec /fluent-bit/bin/fluent-bit -e /fluent-bit/firehose.so -e /fluent-bit/cloudwatch.so -e /fluent-bit/kinesis.so"
+
+// global s3 client and flag
+var s3Client *s3.S3
+var exists3Client bool = false
+
+// global ecs metadata region
+var metadataReigon string = ""
+
+// set error log format
+func setErrorLog() {
+	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
+	log.SetPrefix("FluentBit Init Process -> ")
+}
+
+// create the invoker.sh
+// which will declare ECS Task Metadata as environment variables
+// and finally invoke Fluent Bit
+func createInvokerFile(filePath string) {
+	invokerFile, err := os.Create(filePath)
+	if err != nil {
+		log.Println(err)
+		log.Fatalln("Cannot create fluent bit invoker file")
+	}
+	defer invokerFile.Close()
+}
+
+type ECSTaskMetadata struct {
+	AWS_REGION          string `json:"AWSRegion"`
+	ECS_CLUSTER         string `json:"Cluster"`
+	ECS_TASK_ARN        string `json:"TaskARN"`
+	ECS_TASK_ID         string `json:"TaskID"`
+	ECS_FAMILY          string `json:"Family"`
+	ECS_LAUNCHTYPE      string `json:"LaunchType"` //If container agent is under version 1.45.0, this env var will be an empty string
+	ECS_REVISION        string `json:"Revision"`
+	ECS_TASK_DEFINITION string `json:"TaskDefinition"`
```

Comment:
you can add a comment here for what it means

Reply:
Yeah, already added